### PR TITLE
Add developer docs and a note about abandonment

### PR DESCRIPTION
**Pull request by timholy**
I hereby retire as maintainer of this package. I have written too many packages to support them all, and since this package has a wide user base it should not be too difficult to find someone to fill my shoes.

To facilitate the transition, this PR adds a whole bunch of developer-oriented documentation. It also posts a note on the README advertising for a new maintainer.

### CODE REVIEW

**Comment by timholy** [comment]
Preview the documentation from this PR: https://juliamath.github.io/Interpolations.jl/previews/PR365/. In particular click on the "Developer documentation." But there are also many new docstrings for internal (and even exported) objects.

**Comment by timholy** [comment]
xref https://discourse.julialang.org/t/interpolations-jl-needs-a-new-maintainer/37454

File: src/b-splines/indexing.jl [hunk 1]
```diff
@@ -51,7 +51,16 @@ end
...
 
+"""
+    weightedindexes(fs, itpflags, nodes, xs)
 
+Compute `WeightedIndex` values for evaluation at the position `xs...`.
+`fs` is a function or tuple of functions indicating the types of index required,
+typically `value_weights`, `gradient_weights`, and/or `hessian_weights`.
+`itpflags` and `nodes` can be obtained from `itpinfo(itp)...`.
+
+See the "developer documenation" for further information.
```
Comment (asinghvi17) [inline src/b-splines/indexing.jl:62]:
```suggestion
See the "developer documentation" for further information.

Comment (timholy) [reply src/b-splines/indexing.jl]: Thanks!

**Comment by tomasaschan** [comment]
It's been a looong time since I did anything in the Julia community, but when I started out - by reaching out to you, @timholy, about adding cubic spline interpolations to Grid.jl - I could not have been given a warmer welcome. You took me under your wings for a while, coached me into building the first versions of this package (I assume all of that code is more or less replaced by now) and taking care of it for a few years, and you took on a large part of the burden when my career path took me elsewhere and I didn't find time to do my part as a maintainer anymore (see #140, ages ago).

**Thank you.**

I just wanted to ensure that got said (with enough emphasis).

**Comment by timholy** [comment]
Thanks @tomasaschan! I've always enjoyed collaborating with you on this package. Cheers to its bright future in new hands!

**Pull request merged**